### PR TITLE
Crosswords fix deep linking

### PR DESCRIPTION
**Pull request by NathanielBennett**
Following on from the aborted https://github.com/guardian/frontend/pull/10772, the removes deep linking _for crosswords only_. 
@johnduffell, @michaelwmcnamara , @PetrK-GNM 

Can you guys cast your eyes over this?

### CODE REVIEW

**Comment by PetrK-GNM** [comment]
Do you guys want to catch up about this at some point? Google will move to Apple's "Universal Links" very soon so this change will probably only have limited impact. We need to look into writing some regexes to exclude crosswords and potentially other pages from universal deep-linking: https://developer.apple.com/library/prerelease/ios/documentation/General/Conceptual/AppSearch/UniversalLinks.html

File: common/test/metadata/MetaDataMatcher.scala [hunk 1]
```diff
@@ -40,4 +40,26 @@ object MetaDataMatcher extends Matchers  {
...
+    status(result) should be(200)
+    val script = body.getElementsByAttributeValue("href", expectedLink)
+    script.size() should be(1)
+  }
+
+  def ensureNoDeepLink(result: Future[Result], nonExistantLink: String) {
+    val body = Jsoup.parseBodyFragment(contentAsString(result))
+    status(result) should be(200)
+    val script = body.getElementsByAttributeValue("href", nonExistantLink)
+    script.size() should be(0)
```
Comment (johnduffell) [inline common/test/metadata/MetaDataMatcher.scala]: testing for the absence of something very specific can be dangerous - if we want no deep link at all we should look for something starting with app:// etc?

Comment (NathanielBennett) [inline common/test/metadata/MetaDataMatcher.scala]: @johnduffell yeah, that's what I am doing. The argument nonExistantLink is the the thing which starts with app://., This method just saves me from having to look for things starting with 'app;//' in four different places. see changes to IndexMetaDataTest above .

Comment (johnduffell) [inline common/test/metadata/MetaDataMatcher.scala]: look like it's checking specifically for "ios-app://$appId/gnmguardian/$articleUrl?contenttype=front&source=google" and not just anything starting "ios-app://".

If we slightly changed the format of the url e.g. change the contenttype to ContentType then this test will always pass regardelss.

If we make a test to check no ios deep link it should really check that.

Having said that, maybe we don't consider it a big risk, but thought I'd let you know!

**Pull request merged**